shard.go

```diff
@@ -180,14 +179,31 @@ func marshalValues(values map[uint8]interface{}) []byte {
 		}
 
 		// Encode value after field id.
-		// TODO: Support non-float types.
 		switch v := v.(type) {
 		case float64:
+			buf = make([]byte, 9)
 			binary.BigEndian.PutUint64(buf[1:9], math.Float64bits(v))
+		case bool:
+			buf = make([]byte, 2)
```

Comment:
I think the `ReadSeries` function is actually cruft left over from stuff that was in there for debugging purposes. The real way to read raw series data is through the query engine, which now actually works.

There's some other spot in the code where you can get a value based on the field ID. At some point someone needs to cast this to the correct data type. That should be done in the query engine before it gets yielded to the mappers.